### PR TITLE
Remove asynchronous field resolvers

### PR DESCRIPTION
**Pull request by Shane32**
Remove:
- AsyncFieldResolver
- AsyncEventStreamResolver
- ScopedAsyncFieldResolver

Change references to those classes to convert Task-based results to ValueTask results

Does not change any field builder methods in this PR.

### CODE REVIEW

**Comment by Shane32** [comment]
@sungam3r2022 this is ready for review

File: src/GraphQL/Types/Composite/ObjectGraphTypeExtensions.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public static class ObjectGraphTypeExtensions
                 Arguments = arguments,
                 ResolvedType = type,
                 Resolver = resolve != null
-                    ? new AsyncFieldResolver<object>(resolve)
+                    ? new FuncFieldResolver<object>(context => new ValueTask<object?>(resolve(context)))
```
Comment (sungam3r2022) [inline src/GraphQL/Types/Composite/ObjectGraphTypeExtensions.cs:62]: Note a comment above: 
>//TODO: v5 - change void to T where T : IObjectGraphType

Small change.

File: src/GraphQL/Resolvers/EventStreamResolver.cs [hunk 2]
```diff
@@ -15,6 +15,17 @@ public EventStreamResolver(Func<IResolveFieldContext, IObservable<TReturnType?>>
             _subscriber = context => new ValueTask<IObservable<object?>>((IObservable<object?>)subscriber(context));
         }
 
+        public EventStreamResolver(Func<IResolveFieldContext, ValueTask<IObservable<TReturnType?>>> subscriber)
+        {
+            if (subscriber == null)
```
Comment (sungam3r2022) [inline src/GraphQL/Resolvers/EventStreamResolver.cs:20]: I'm waiting for C#7 and new syntax:
`  public EventStreamResolver(Func<IResolveFieldContext, ValueTask<IObservable<TReturnType?>>>!! subscriber)`

`!!` means that compiler generates null checks

File: src/GraphQL.MicrosoftDI/ScopedFieldResolver.cs [hunk 3]
```diff
@@ -42,5 +52,17 @@ public class ScopedFieldResolver<TSourceType, TReturnType> : FuncFieldResolver<T
                 return resolver(new ScopedResolveFieldContextAdapter<TSourceType>(context, scope.ServiceProvider));
             };
         }
+
+        /// <inheritdoc cref="ScopedFieldResolver{TSourceType, TReturnType}.ScopedFieldResolver(Func{IResolveFieldContext{TSourceType}, TReturnType})" />
+        public ScopedFieldResolver(Func<IResolveFieldContext<TSourceType>, ValueTask<TReturnType?>> resolver) : base(GetScopedResolver(resolver)) { }
+
+        private static Func<IResolveFieldContext, ValueTask<TReturnType?>> GetScopedResolver(Func<IResolveFieldContext<TSourceType>, ValueTask<TReturnType?>> resolver)
+        {
+            return async (context) =>
```
Comment (sungam3r2022) [inline src/GraphQL.MicrosoftDI/ScopedFieldResolver.cs:61]:
```suggestion
            return async context =>
```

File: src/GraphQL.MicrosoftDI/ScopedFieldResolver.cs [hunk 4]
```diff
@@ -14,24 +14,34 @@ public class ScopedFieldResolver<TReturnType> : FuncFieldResolver<TReturnType>
...
                 return resolver(new ScopedResolveFieldContextAdapter<object>(context, scope.ServiceProvider));
             };
         }
+
+        /// <inheritdoc cref="ScopedFieldResolver{TReturnType}.ScopedFieldResolver(Func{IResolveFieldContext, TReturnType})"/>
+        public ScopedFieldResolver(Func<IResolveFieldContext, ValueTask<TReturnType?>> resolver) : base(GetScopedResolver(resolver)) { }
+
+        private static Func<IResolveFieldContext, ValueTask<TReturnType?>> GetScopedResolver(Func<IResolveFieldContext, ValueTask<TReturnType?>> resolver)
+        {
+            return async (context) =>
```
Comment (sungam3r2022) [inline src/GraphQL.MicrosoftDI/ScopedFieldResolver.cs:33]:
```suggestion
            return async context =>
```

File: src/GraphQL.DataLoader/DataLoaderExtensions.cs [hunk 5]
```diff
@@ -132,6 +132,6 @@ async Task<object> IDataLoaderResult.GetResultAsync(CancellationToken cancellati
 
         /// <inheritdoc cref="FieldBuilder{TSourceType, TReturnType}.Resolve(IFieldResolver)"/>
         public static FieldBuilder<TSourceType, TReturnType> ResolveAsync<TSourceType, TReturnType>(this FieldBuilder<TSourceType, TReturnType> builder, Func<IResolveFieldContext<TSourceType>, Task<IDataLoaderResult<IDataLoaderResult<IDataLoaderResult<TReturnType>>>>> resolve)
-            => builder.Resolve(new AsyncFieldResolver<TSourceType, IDataLoaderResult<IDataLoaderResult<IDataLoaderResult<TReturnType>>>>(resolve));
+            => builder.Resolve(new FuncFieldResolver<TSourceType, IDataLoaderResult<IDataLoaderResult<IDataLoaderResult<TReturnType>>>>(context => new ValueTask<IDataLoaderResult<IDataLoaderResult<IDataLoaderResult<TReturnType>>>>(resolve(context))));
```
Comment (sungam3r2022) [inline src/GraphQL.DataLoader/DataLoaderExtensions.cs:135]: 😄

**Review by sungam3r2022 — APPROVED** [review]
I am glad that this part of the system was so simplified.

File: src/GraphQL/Types/Composite/ObjectGraphTypeExtensions.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL/Types/Composite/ObjectGraphTypeExtensions.cs:62]: #2998

**Pull request merged**